libs/agent-scripts/libvirt/createtemplate.py

```diff
@@ -15,24 +14,93 @@
 async = True
 
 
-def action(ovs_connection, diskguid):
+def action(ovs_connection, diskguid, new_vdiskguid, template_name):
     # Creates sets vdisk as a template
     #
     # ovs_connection: dict holding connection info for ovs restapi
     #   eg: { ips: ['ip1', 'ip2', 'ip3'], client_id: 'dsfgfs', client_secret: 'sadfafsdf'}
     # diskguid: disk of which we are creating a template
     #
     # returns diskguid of the created template
+    from CloudscalerLibcloud import openvstorage
+    from CloudscalerLibcloud.utils.gridconfig import GridConfig
+    config = GridConfig()
+    username = config.settings['ovs_credentials']['edgeuser']
+    password = config.settings['ovs_credentials']['edgepassword']
 
     ovs = j.clients.openvstorage.get(ips=ovs_connection['ips'],
                                      credentials=(ovs_connection['client_id'],
                                                   ovs_connection['client_secret']))
+    disk = ovs.get('/vdisks/{}'.format(diskguid))
+
+    storage_drivers = ovs.get('/storagedrivers', params={'contents': 'storagedriver_id'})['data']
+    def getDiskPath(disk):
+        storagedriver_id = disk['storagedriver_id']
+        for sd in storage_drivers:
+            if sd['storagedriver_id'] == storagedriver_id:
+                break
+        storage_ip = sd['storage_ip']
+        edge_port = sd['ports']['edge']
+        path = 'openvstorage+tcp:{}:{}{}:username={}:password={}'.format(storage_ip, edge_port, disk['devicename'].split('.')[0], username, password)
```

Comment:
please make sure it also works for OVS OSE (when not password and username is set)